### PR TITLE
feat: Added profile parameter

### PR DESCRIPTION
**Pull request by gonzariosm**
*Changes:*

Added the `--profile` option to pass a different profile if you manage multiple AWS accounts.

### CODE REVIEW

**Comment by jakob-vendegna-sp** [comment]
FYI I filed a support case about this with AWS. It seems the maintainer of this repo is out on an extended leave with no defined return date. So... I guess this is on-hold until further notice. 😢

**Comment by jakob-vendegna-sp** [comment]
Oh, but it wouldn't hurt to purge those pycache files in your PR in the meantime. I know they'll ask about that first.

File: capacity-mode-evaluator/src/pricing.py [hunk 1]
```diff
@@ -5,7 +5,7 @@
 
 
 class PricingUtility(object):
-    def __init__(self, region_name):
+    def __init__(self, region_name, profile_name='default'):
```
Comment (tebanieo) [inline capacity-mode-evaluator/src/pricing.py:8]: This change will force everyone to use a profile, there are customers that use this tool with environment variables.

File: capacity-mode-evaluator/src/pricing.py [hunk 2]
```diff
@@ -17,8 +17,9 @@ def __init__(self, region_name):
         # pick the closest endpoint to the supplied region
         if region_name not in AMERICAN_REGIONS:
             closest_api_region = 'ap-south-1'
-
-        self.pricing_client = boto3.client(
+        
+        self.session = boto3.session.Session(profile_name=profile_name)
```
Comment (tebanieo) [inline capacity-mode-evaluator/src/pricing.py:21]: Implement the necessary conditional and modify the code to accommodate None as an input for customers utilizing environment variables.

File: table_tagger.py [hunk 3]
```diff
@@ -88,6 +88,9 @@ def main():
 
     parser.add_argument(
         '--tag-name', required=False, type=str, default='table_name',  help='tag table with tag TAG_NAME (default is "table_name")')
+    
+    parser.add_argument(
+        '--profile', required=False, type=str, default='default', help='set a custom profile name to perform the operation under')
```
Comment (tebanieo) [inline table_tagger.py:93]: The default value should be none, what if you want to use the environment variables instead of the client?

**Review by tebanieo — CHANGES_REQUESTED** [review]
Hello Thanks for the PR, merging this code will add a strong dependency on having set up your profile, there are some situations where customers work with environment variables instead of the profile file. Please make this optional as we don't want to force anyone to use the default `.profile` file.

File: ddbtools/__pycache__/constants.cpython-311.pyc
Comment (tebanieo) [inline ddbtools/__pycache__/constants.cpython-311.pyc:1]: add `*.pyc` to your .git ignore so you don't push the pycache.

File: capacity-mode-evaluator/src/pricing.py (hunk 1 above)
Comment (gonzariosm) [reply capacity-mode-evaluator/src/pricing.py]: Updated to `None`

File: capacity-mode-evaluator/src/pricing.py (hunk 2 above)
Comment (gonzariosm) [reply capacity-mode-evaluator/src/pricing.py:21]: Updated to `None`

File: table_tagger.py (hunk 3 above)
Comment (gonzariosm) [reply table_tagger.py]: Updated to `None`

File: ddbtools/__pycache__/constants.cpython-311.pyc
Comment (gonzariosm) [reply ddbtools/__pycache__/constants.cpython-311.pyc:1]: Added some lines in `.gitignore`and removed `__pycache__`

**Comment by gonzariosm** [comment]
@tebanieo can you take a look and let me know if I can add something else to this PR, please?

**Review by tebanieo — APPROVED** [review]

**Pull request merged**